docs/execution-providers/CUDA-ExecutionProvider.md

```diff
@@ -57,10 +57,26 @@ The device ID.
 
 Default value: 0
 
+### has_user_compute_stream
+Defines that the user has set a `user_compute_stream`.
+This cannot be used in combination with an external allocator.
```

Comment:
@hariharans29 I am a little conflicted with the legacy CUDA options and the (preferred) V2 API. While I can actually specify that the session has a user compute stream through V2 API it's not possible to specify the actual stream. What's the guidance on this if I need some options that are only available through V2 API, but at the same time I need to specify e.g. the compute stream ?

Reply:
Over the Python API using `onnxruntime.InferenceSession` it is not possible to set `user_compute_stream`, as it throws:

```
EP Error /onnxruntime_src/onnxruntime/core/providers/cuda/cuda_execution_provider_info.cc:51 static onnxruntime::CUDAExecutionProviderInfo onnxruntime::CUDAExecutionProviderInfo::FromProviderOptions(const ProviderOptions&) [ONNXRuntimeError] : 1 : FAIL : provider_options_utils.h:146 Parse Unknown provider option: "user_compute_stream".
 when using [('CUDAExecutionProvider', {'has_user_compute_stream': True, 'user_compute_stream': <pycuda._driver.Stream object at 0x7f62e375d0d0>})]
Falling back to ['CUDAExecutionProvider', 'CPUExecutionProvider'] and retrying.
```
on
```
session = onnxruntime.InferenceSession(path, opts, providers=[("CUDAExecutionProvider", 
                    {
                        'has_user_compute_stream': True,
                        'user_compute_stream': stream 
                    })])
```

Reply:
user_compute_stream is not available in python API.

Reply:
I see it as a very important setting. Are there plans to expose this ?